src/SeisSol.cpp

```diff
@@ -98,14 +99,19 @@ bool seissol::SeisSol::init(int argc, char* argv[]) {
                 << parallel::Pinning::maskToString(pinning.getWorkerUnionMask());
   logInfo(rank) << "OpenMP worker affinity (this node)   :"
                 << parallel::Pinning::maskToString(pinning.getNodeMask());
-#endif
-#ifdef USE_COMM_THREAD
-  auto freeCpus = pinning.getFreeCPUsMask();
-  logInfo(rank) << "Communication thread affinity        :"
-                << parallel::Pinning::maskToString(freeCpus);
-  if (parallel::Pinning::freeCPUsMaskEmpty(freeCpus)) {
-    logError()
-        << "There are no free CPUs left. Make sure to leave one for the communication thread.";
+
+  seissol::printCommThreadInfo(MPI::mpi);
+  if (seissol::useCommThread(MPI::mpi)) {
+    auto freeCpus = pinning.getFreeCPUsMask();
+    logInfo(rank) << "Communication thread affinity        :"
+                  << parallel::Pinning::maskToString(freeCpus);
+    if (parallel::Pinning::freeCPUsMaskEmpty(freeCpus)) {
+      logError()
+          << "There are no free CPUs left. Make sure to leave one for the communication thread. If "
+             "you want to run SeisSol without a communication thread (and instead use polling), "
+             "then try running with the environment variable \"SEISSOL_COMMTHREAD=false\". "
+             "However, for more balanced performance, we recommend using the communication thread.";
```

Comment:
what means more balanced performance? higher performance?

Reply:
Removed that sentence.